python/lsst/daf/butler/registry/interfaces/_versioning.py

```diff
@@ -180,19 +180,62 @@ def newSchemaVersion(self) -> VersionTuple | None:
 
         Notes
         -----
-        Default implementation only forks for extensions that support single
-        schema version and it returns version obtained from `currentVersions`.
-        If `currentVersions` returns multiple version then default
-        implementation will raise an exception and the method has to be
-        reimplemented by a subclass.
+        Extension classes which support multiple schema versions need to
+        override `_newDefaultSchemaVersion` method.
         """
-        my_versions = self.currentVersions()
+        return self.clsNewSchemaVersion(self._registry_schema_version)
+
+    @classmethod
+    def clsNewSchemaVersion(cls, schema_version: VersionTuple | None) -> VersionTuple | None:
+        """Class method which returns schema version for new registry.
```

Comment:
Could you clarify whether this means a new registry database vs. just a new registry client interface?